classes/sqliimportitem.php

```diff
@@ -163,8 +163,17 @@ public function __get( $name )
     public function getOptions()
     {
         if ( !$this->options instanceof SQLIImportHandlerOptions && $this->attribute( 'options_serialized' ) )
-            $this->options = unserialize( $this->attribute( 'options_serialized' ) );
-            
+        {
+            if ( eZINI::instance()->variable( 'DatabaseSettings', 'DatabaseImplementation' ) == 'ezpostgresql' )
+            {
+                $safeSerialized = str_replace( "~~NULL_BYTE~~", "\0", $this->attribute( 'options_serialized' ) );
+                $this->options = unserialize( $safeSerialized );
+            }
+            else
+            {
+                $this->options = unserialize( $this->attribute( 'options_serialized' ) );
+            }
```

Comment:
As this block is repeated many times, maybe it's worth factor it somewhere? Like in SQLIImportUtils?